sdks/python/apache_beam/io/gcp/dicomio_integration_test.py

```diff
@@ -0,0 +1,155 @@
+  # Delete an existing DICOM store
+  credential, _ = default()
+  session = requests.AuthorizedSession(credential)
+  api_endpoint = "{}/projects/{}/locations/{}".format(
+      HEALTHCARE_BASE_URL, project_id, region)
+
+  # base of dicomweb path.
+  dicomweb_path = "{}/datasets/{}/dicomStores/{}".format(
+      api_endpoint, dataset_id, dicom_store_id)
+
+  response = session.delete(dicomweb_path)
+  response.raise_for_status()
+  return response.status_code
+
+
+class DICOMIoIntegrationTest(unittest.TestCase):
+  expected_output_metadata = None
+
+  def setUp(self):
+    self.test_pipeline = TestPipeline(is_integration_test=True)
+    self.project = self.test_pipeline.get_option('project')
+
+    # create a temp Dicom store based on the time stamp
+    self.temp_dicom_store = "DICOM_store_" + str(time.time())
+    create_dicom_store(self.project, DATA_SET_ID, REGION, self.temp_dicom_store)
+    client = storage.Client()
+    bucket = client.get_bucket('temp-storage-for-dicom-io-tests')
+    blob = bucket.blob('meta_data_json/Dicom_io_it_test_data.json')
+    data = json.loads(blob.download_as_string())
+    self.expected_output_metadata = data
+
+  def tearDown(self):
+    # clean up the temp Dicom store
+    delete_dicom_store(self.project, DATA_SET_ID, REGION, self.temp_dicom_store)
+
+  @attr('IT')
+  def test_dicom_search(self):
+    # Search and compare the metadata of a persistent DICOM store.
+    input_dict = {}
+    input_dict['project_id'] = self.project
+    input_dict['region'] = REGION
+    input_dict['dataset_id'] = DATA_SET_ID
+    input_dict['dicom_store_id'] = PERSISTENT_DICOM_STORE_NAME
+    input_dict['search_type'] = "instances"
+
+    expected_dict = {}
+    expected_dict['result'] = self.expected_output_metadata
+    expected_dict['status'] = 200
+    expected_dict['input'] = input_dict
+    expected_dict['success'] = True
+
+    with TestPipeline() as p:
+      results = (p | beam.Create([input_dict]) | DicomSearch())
+      assert_that(results, equal_to([expected_dict]))
+
+  @attr('IT')
+  def test_dicom_store_instance(self):
+    # Store DICOM files to a empty DICOM store from a GCS bucket,
+    # then check if the store metadata match.
+    input_dict = {}
+    input_dict['project_id'] = self.project
+    input_dict['region'] = REGION
+    input_dict['dataset_id'] = DATA_SET_ID
+    input_dict['dicom_store_id'] = self.temp_dicom_store
+    input_dict['search_type'] = "instances"
+
+    expected_dict = {}
+    expected_dict['result'] = self.expected_output_metadata
+    expected_dict['status'] = 200
+    expected_dict['input'] = input_dict
+    expected_dict['success'] = True
+
+    with TestPipeline() as p:
+      gcs_path = DICOM_FILES_PATH + "/*"
+      results = (
+          p
+          | fileio.MatchFiles(gcs_path)
+          | fileio.ReadMatches()
+          | UploadToDicomStore(input_dict, 'fileio')
+          | beam.Map(lambda x: x['success']))
+      assert_that(results, equal_to([True] * NUM_INSTANCE))
```

Comment:
We should define the expected value above and not inline.

Reply:
Changed